libs/dh/admin/data-access-api/src/lib/dh-admin-user-management-data-access-api.store.ts

```diff
@@ -176,23 +185,32 @@ export class DhAdminUserManagementDataAccessApiStore
     statusFilter,
     actorIdFilter,
     userRoleFilter,
-  }: FetchUsersParams) {
+  }: FetchUsersParams): Observable<UserResponseType> {
     if (!statusFilter || statusFilter.length == 0) {
-      return of({ users: [], totalUserCount: 0 });
+      return of({ data: null, loading: false, error: false });
     }
 
-    return this.httpClient.v1MarketParticipantUserOverviewSearchUsersPost(
-      pageNumber,
-      pageSize,
-      sortProperty,
-      direction,
-      {
-        actorId: actorIdFilter,
-        userRoleIds: userRoleFilter,
-        searchText: searchText,
-        userStatus: statusFilter,
-      }
-    );
+    return this.apollo
+      .query({
+        query: UserOverviewSearchDocument,
+        variables: {
+          pageNumber,
+          pageSize: pageSize,
+          sortProperty,
+          sortDirection: direction,
+          actorId: actorIdFilter,
+          userRoleIds: userRoleFilter,
+          searchText,
+          userStatus: statusFilter,
+        },
+      })
+      .pipe(
+        map((response) => ({
+          loading: response.loading,
+          error: Boolean(response.error) || (response.errors?.length ?? 0) > 0,
+          data: response.data?.userOverviewSearch,
```

Comment:
suggestion: should probably fallback to an empty array otherwise it will end up being `undefined`


```suggestion
          data: response.data?.userOverviewSearch ?? [],
```